### DISCUSSION TITLE
Question: Is there a way to get a RPG column guide across the top?

### DISCUSSION THREAD

**Discussion by sprinkles** [codefori/vscode-ibmi#162] [Ideas]
I realize that this is mostly for writing in RPGLE, but for files that are still in RPG... Is there any way to get a line across the top showing a column guide? There might be a setting that I've missed, or this might just be something that's not included. I'm just curious. Thanks!

**Comment by worksofliam** [comment]
Great question. Not right now... but it's something a codelens might be able to help with.

Let me look into it!

**Reply by worksofliam** [reply]
@sprinkles By any chance, are you able to provide links to the documentation which specify each spec (C, D, H, etc) and what their different positions represent?

**Reply by sprinkles** [reply]
Thanks for the response @worksofliam! The only documentation I would have is just off the IBM site. I'm fairly new to using RPG. I prefer RPG free, but I'm working with a lot of old programs. My office doesn't have a license for RDi. We've been using an old copy of websphere (which has the column guide at the top), but it constantly crashes on my computer. I've been using this as an alternative, but having that column guide at the top could be a a great help in this tool. 
This is a screenshot of the guide for an RPGLE program, but it changes depending on the program, of course. 
![Screenshot 2021-06-11 111558](https://user-images.githubusercontent.com/849665/121721532-f8f4ab00-caa9-11eb-80cb-ab585492f499.png)

**Reply by worksofliam** [reply]
@sprinkles Is there any chance you can link me to any of those docs?

**Reply by worksofliam** [reply]
I found it:

* https://www.ibm.com/docs/en/i/7.4?topic=reference-specifications

I will use this as a base when building this part of the extension.

**Comment by worksofliam** [comment]
This is my progress:

![image](https://user-images.githubusercontent.com/3708366/121760158-519e6500-caf7-11eb-948f-3ec8805e18f7.png)

Basically, it will highlight the different parts of the spec and also allow you to highlight them. There isn't a clear way to display text right now... so the best I could do is highlight and also add some hover text.

I will post more as I go.

**Reply by worksofliam** [reply]
Column prompter
![image](https://user-images.githubusercontent.com/3708366/121785660-1482b300-cb89-11eb-8a08-a97e739a6a54.png)

**Comment by worksofliam** [comment]
@sprinkles I don't think there is any way this functionality is going to make it into master for a little bit. I am going to make a PR and it's going to need a lot of feedback before we release it.

If I make a PR, would you be able to build the extension yourself and have a play around with its functionality?

**Reply by sprinkles** [reply]
@worksofliam I haven't before, but I could try it. Also, I like what you've done! It looks fun.

**Reply by sprinkles** [reply]
Hi @worksofliam -- I'm in the process of switching jobs right now and no longer have access to the IBM i that I was working on, so I won't be able to test anything or work on any of this for the time being. I just wanted to let you know. Thank you so much for working on this with me though!

**Reply by worksofliam** [reply]
That is sad to hear. Thank you for letting me know.

**Comment by CeGarrett** [comment]
I currently use RDI but, looking at Code4i after TechXchange. 

It seems that the coloring of the columns made it into the code. Did the Column Assistant get added? How do I enable/display it?
